src/structures/Role.js

```diff
@@ -82,6 +82,44 @@ class Role extends Base {
      * @type {boolean}
      */
     this.deleted = false;
+
+    /**
+     * The tags this role has
+     * @type {?Object}
+     * @property {?Snowflake} botID The id of the bot this role belongs to
+     * @property {?Snowflake} integrationID The id of the integration this role belongs to
+     * @property {?boolean} premiumSubcriberRole Whether this is the guilds premium subscription role
+     */
+    this.tags = data.tags ? {} : null;
+    if (data.tags) {
+      if (Reflect.has(data.tags, 'bot_id')) {
+        this.tags.botID = data.tags.bot_id;
+      }
+      if (Reflect.has(data.tags, 'integration_id')) {
+        this.tags.integrationID = data.tags.integration_id;
+      }
+      if (Reflect.has(data.tags, 'premium_subscriber')) {
+        this.tags.premiumSubscriberRole = true;
+      }
+    }
+  }
+
+  /**
+   * Whether this role is the guilds premium subscription role
+   * @type {boolean}
+   * @readonly
+   */
+  get isPremiumSubscriberRole() {
+    return Boolean(this.tags && this.tags.premiumSubscriberRole);
+  }
+
+  /**
+   * Whether this role is the role associated with the logged in bot user
```

Comment:
```suggestion
   * Whether this role is the role associated with the logged in client user
```
Should be made consistent with [this](https://github.com/discordjs/discord.js/pull/4628/files#diff-1e631e07c0e6467138781b3eed92f077R145)